binding/ios/PorcupineManager.swift

```diff
@@ -30,28 +30,7 @@ public class PorcupineManager {
         self.onDetection = onDetection
         self.porcupine = porcupine
         
-        self.audioInputEngine = AudioInputEngine()
-        audioInputEngine.audioInput = { [weak self] audio in
-            
-            guard let `self` = self else {
-                return
-            }
-            
-            guard self.porcupine != nil else {
-                return
-            }
-            
-            do{
-                let result:Int32 = try self.porcupine!.process(pcm: audio)
-                if result >= 0 {
-                    DispatchQueue.main.async {
-                        self.onDetection?(result)
-                    }
-                }
-            } catch {
-                print("\(error)")
-            }
-        }
+        self.voiceProcessor = VoiceProcessor()
```

Comment:
our other voice processors all conform to the singleton pattern. I think you should try to do the same with ios-voice-processor. Reference the flutter-voice-processor for how that's done. If we do instances, then a user could potentially create several voice processors, which would not function properly.